### PR TITLE
Added requirements file to config zip and GUI view of requirements

### PR DESCRIPTION
**Pull request by vinceli1004**
This PR adds the ability to add a requirements.txt file for custom challenge package installation.

Approach:
The submission worker, upon extracting the challenge configuration zip, checks for a requirements.txt file and attempts to install its contents with a subprocess call. Any issues are logged.

An API was added to get the requirements given the challenge id, and this was used to display the requirements on a challenge's manage page.

### CODE REVIEW

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]
@vinceli1004 Can you please add the description for the PR mainly what this PR adds and what is the approach for it.

**Comment by vinceli1004** [comment]
> @vinceli1004 Can you please add the description for the PR mainly what this PR adds and what is the approach for it.

Added!

File: apps/challenges/migrations/0086_challenge_requirements.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 2.2.20 on 2021-08-11 06:51
```
Comment (Ram81) [inline apps/challenges/migrations/0086_challenge_requirements.py:1]: @vinceli1004 we should remove this migration file as we are not making any model changes

File: apps/challenges/migrations/0087_auto_20210811_0825.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 2.2.20 on 2021-08-11 08:25
```
Comment (Ram81) [inline apps/challenges/migrations/0087_auto_20210811_0825.py:1]: same here

File: apps/challenges/migrations/0088_auto_20210811_0833.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 2.2.20 on 2021-08-11 08:33
```
Comment (Ram81) [inline apps/challenges/migrations/0088_auto_20210811_0833.py:1]: same here

File: apps/challenges/migrations/0089_auto_20210811_0842.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 2.2.20 on 2021-08-11 08:42
```
Comment (Ram81) [inline apps/challenges/migrations/0089_auto_20210811_0842.py:1]: same here

File: apps/challenges/migrations/0090_auto_20210813_0245.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 2.2.20 on 2021-08-13 02:45
```
Comment (Ram81) [inline apps/challenges/migrations/0090_auto_20210813_0245.py:1]: same here

File: apps/challenges/migrations/0091_remove_challenge_requirements.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# Generated by Django 2.2.20 on 2021-09-04 23:30
```
Comment (Ram81) [inline apps/challenges/migrations/0091_remove_challenge_requirements.py:1]: same here

File: apps/challenges/models.py [hunk 7]
```diff
@@ -73,6 +73,7 @@ def __init__(self, *args, **kwargs):
     evaluation_script = models.FileField(
         default=False, upload_to=RandomFileName("evaluation_scripts")
     )  # should be zip format
+
```
Comment (Ram81) [inline apps/challenges/models.py:76]: remove this newline

File: apps/challenges/views.py [hunk 8]
```diff
@@ -2603,6 +2603,60 @@ def get_challenge_phases_by_challenge_pk(request, challenge_pk):
...
+    extract_location = join(
+        base_location, "data{}".format(challenge_pk)
+    )
+
+    zip_ref = zipfile.ZipFile(challenge.evaluation_script, "r")
+    zip_ref.extractall(extract_location)
+    zip_ref.close()
+    try:
+        os.remove(zip_location)
+    except:
```
Comment (Ram81) [inline apps/challenges/views.py:2632]: Please handle the exception using a exception class

File: apps/challenges/views.py [hunk 9]
```diff
@@ -2603,6 +2603,60 @@ def get_challenge_phases_by_challenge_pk(request, challenge_pk):
...
+
+    if os.path.isfile(requirements_location):
+        f = open(requirements_location, "r")
+        requirement_data = f.read()
+        requirement_data = requirement_data.splitlines()
+        f.close()
+
+    try:
+        shutil.rmtree(base_location)
+    except:  # noqa: E722
```
Comment (Ram81) [inline apps/challenges/views.py:2649]: same here

File: scripts/workers/submission_worker.py [hunk 10]
```diff
@@ -277,6 +279,16 @@ def extract_challenge_data(challenge, phases):
         evaluation_script_url, challenge_zip_file, challenge_data_directory
     )
 
+    try:
+        requirements_location = join(challenge_data_directory, "requirements.txt")
+        if os.path.isfile(requirements_location):
+            subprocess_output = subprocess.check_output([sys.executable, "-m", "pip", "install", "-r", requirements_location])
+            print(subprocess_output)
```
Comment (Ram81) [inline scripts/workers/submission_worker.py:286]: Please remove the print statement

**Review by Ram81 — CHANGES_REQUESTED** [review]

File: apps/challenges/urls.py [hunk 11]
```diff
@@ -151,6 +151,11 @@
         views.get_challenge_phases_by_challenge_pk,
         name="get_challenge_phases_by_challenge_pk",
     ),
+    url(
```
Comment (Ram81) [inline apps/challenges/urls.py:154]: @vinceli1004 can we remove this API

**Review by Ram81 — CHANGES_REQUESTED** [review]
@vinceli1004 can we remove the API based on our discussion.

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**